MSVC/gen-build-config.bat

```diff
@@ -136,36 +149,37 @@ EXIT /B %ERRORLEVEL%
 
 :: Obtain the git version information
 :GET_GIT_VERSION
-	FOR /F "usebackq tokens=*" %%V IN (`git describe --long "--match=[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]" --dirty=-WIP --abbrev=8`) DO (
+
+	FOR /F "usebackq tokens=*" %%V IN (`git describe --long "--match=[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]" "--dirty=-WIP" "--abbrev=8"`) DO (
 		SET $FULL_VERSION=%%V
 	)
-	FOR /F "tokens=1 delims=-" %%M in (%$FULL_VERSION%) DO (
-		SET $MAJOR_VERSION=%%M
-	)
-	FOR /F "tokens=1,* delims=-" %%M in (%$FULL_VERSION%) DO (
-		SET $MINOR_VERSION=%%M
+	FOR /F "tokens=1 delims=-" %%M in ("%$FULL_VERSION%") DO (
+		SET $VERSION_MAJOR=%%M
 	)
-	FOR /F "tokens=1 delims=-" %%M in (%$MINOR_VERSION%) DO (
+	FOR /F "tokens=2 delims=-" %%M in ("%$FULL_VERSION%") DO (
 		SET $MINOR_VERSION_COMMITS=%%M
 	)
-	FOR /F "tokens=2 delims=-" %%M in (%$MINOR_VERSION%) DO (
+	FOR /F "tokens=3 delims=-" %%M in ("%$FULL_VERSION%") DO (
 		SET $MINOR_VERSION_SHA1=%%M
 	)
-	FOR /F "tokens=2 delims=-" %%M in (%$MINOR_VERSION%) DO (
+	FOR /F "tokens=4 delims=-" %%M in ("%$FULL_VERSION%") DO (
 		SET $MINOR_VERSION_WIP=%%M
-	)
+	)	
+	SET $VERSION_MINOR=%$MINOR_VERSION_COMMITS%-%$MINOR_VERSION_SHA1%-%$MINOR_VERSION_WIP%
```

Comment:
It did not work, there was somehow an error with $MINOR_VERSION_SHA which I did not really understand why it happens, but I can play around with that lines again, if you want